API design: Use browser's progress bar to download software package (#19751)
fleetdm/fleet: docs/REST API/rest-api.md

`Status: 200`

### Get or download software

_Available in Fleet Premium._

Get info or download a software package.

`GET /api/v1/fleet/software/titles/:software_title_id/package`

#### Parameters

| Name            | Type    | In   | Description                                      |
| ----            | ------- | ---- | --------------------------------------------     |
| software_title_id              | integer | path | **Required**. The ID of the software title to download software package.|
| team_id | integer | form | **Required**. The team ID. Downloads a software package added to the specified team. |

> | alt             | integer | query | If specified and set to `media`, downloads the specified software package. |

Reviewer: ```suggestion
| alt             | string | query | If specified and set to `media`, downloads the specified software package. If set to `token` the return is a one time use token to download this package unauthenticated. |
```

| token | string | query | Download with authentication token. `alt` must be specified as `media` to download with the token. |

#### Example (get package info)

`GET /api/v1/fleet/software/titles/123/package`

##### Default response

`Status: 200`

```json
{
  "name": "ZoomFullInstaller.pkg"

>   "token": "0331f58a-5bf8-4eb6-959d-f1dbd7fa8d64"

Reviewer: token shouldn't live in info. It's a short lived one time use secret so I think it's more appropriate to only get a token when requesting with `alt=token` IMO

Reviewer: We don't want every get info to store a token that may not be needed.

_Available in Fleet Premium._

Get info or download a software package.

`GET /api/v1/fleet/software/titles/:software_title_id/package`

#### Parameters

| Name            | Type    | In   | Description                                      |
| ----            | ------- | ---- | --------------------------------------------     |
| software_title_id              | integer | path | **Required**. The ID of the software title to download software package.|
| team_id | integer | form | **Required**. The team ID. Downloads a software package added to the specified team. |

> | alt             | integer | query | **Required**. Supported values are `media` and `token`. If set to `media`, downloads the specified software package. If set to `token` returns a one-time use token to download this package unauthenticated. |

Reviewer: ```suggestion
| alt             | integer | query | **Required**. If specified and set to "media", downloads the specified software package. |
```

| token | string | query | Token to download unauthenticated. `alt` must be specified as `media` to download with the token. |

#### Example (`alt=media`)

> `GET /api/v1/fleet/software/titles/123/package?alt=media?team_id=2`

Reviewer: ```suggestion
`GET /api/v1/fleet/software/titles/123/package?alt=media&team_id=2`
```

Body: <blob>
```

> #### Example (`alt=token`)
> 
> `GET /api/v1/fleet/software/titles/123/package?alt=token?team_id=2`
> 
> ##### Default response
> 
> `Status: 200`
> 
> ```json
> {
>   "token": "0331f58a-5bf8-4eb6-959d-f1dbd7fa8d64"
> }
> ```
> 

Reviewer: ```suggestion
```
